cmd/kubeadm/app/discovery/token/token.go

```diff
@@ -74,16 +74,11 @@ func RetrieveValidatedConfigInfo(cfg *kubeadmapi.JoinConfiguration) (*clientcmda
 		klog.V(1).Infof("[discovery] Created cluster-info discovery client, requesting info from %q\n", insecureBootstrapConfig.Clusters[clusterName].Server)
 
 		// Make an initial insecure connection to get the cluster-info ConfigMap
-		var insecureClusterInfo *v1.ConfigMap
-		wait.PollImmediateInfinite(constants.DiscoveryRetryInterval, func() (bool, error) {
-			var err error
-			insecureClusterInfo, err = insecureClient.CoreV1().ConfigMaps(metav1.NamespacePublic).Get(bootstrapapi.ConfigMapClusterInfo, metav1.GetOptions{})
-			if err != nil {
-				klog.V(1).Infof("[discovery] Failed to request cluster info, will try again: [%s]\n", err)
-				return false, nil
-			}
-			return true, nil
-		})
+		insecureClusterInfo, err := insecureClient.CoreV1().ConfigMaps(metav1.NamespacePublic).Get(bootstrapapi.ConfigMapClusterInfo, metav1.GetOptions{})
```

Comment:
like i've mentioned in the issue i think we should still PollImmediate with a sensible timeout:

possibly also change this infinite poll to the same to be consistent:
```
klog.V(1).Infof("[discovery] Requesting info from %q again to validate TLS against the pinned public key\n", 
...
wait.PollImmediateInfinite(constants.DiscoveryRetryInterval, func() (bool, error) {
```

@chuckha 
LMK if you are out of time for today and i can send the change?

Reply:
actually. i think i might be misunderstanding what is going here... :thinking:

Reply:
ok, so we need to change the second `PollImmediateInfinite` to a `CoreV1().ConfigMaps(metav1.NamespacePublic).Get` too.

Reply:
Ah yes, there is another poll infinite in this function, that needs to be removed as well.

The reason I believe these should be removed is that the `wait.Until` that calls this function is being run every 5 seconds (by default). That makes me think this function should not have any polling/retry logic as the retry logic is scoped to the parent function.

Reply:
I would have taken the reverse approach - narrow down the retries to the actual connecting funcs. But I think, that this is good enough for the time being.